services/src/datasets/external/netcdfcf/schema/migration_0000_initial.sql

```diff
@@ -0,0 +1,248 @@
+);
+
+CREATE TYPE "SpatialReferenceAuthority" AS ENUM (
+    'Epsg',
+    'SrOrg',
+    'Iau2000',
+    'Esri'
+);
+
+CREATE TYPE "SpatialReference" AS (
+    authority "SpatialReferenceAuthority",
+    code OID
+);
+
+CREATE TYPE "Coordinate2D" AS (
+    x double precision,
+    y double precision
+);
+
+CREATE TYPE "TimeInterval" AS (
+    start bigint,
+    "end" bigint
+);
+
+CREATE TYPE "SpatialResolution" AS (
+    x double precision,
+    y double precision
+);
+
+CREATE TYPE "ContinuousMeasurement" AS (
+    measurement text,
+    unit text
+);
+
+CREATE TYPE "SmallintTextKeyValue" AS (
+    key smallint,
+    value text
+);
+
+CREATE TYPE "TextTextKeyValue" AS (
+    key text,
+    value text
+);
+
+CREATE TYPE "ClassificationMeasurement" AS (
+    measurement text,
+    classes "SmallintTextKeyValue" []
+);
+
+CREATE TYPE "Measurement" AS (
+    -- "unitless" if all none
+    continuous "ContinuousMeasurement",
+    classification "ClassificationMeasurement"
+);
+
+CREATE TYPE "SpatialPartition2D" AS (
+    upper_left_coordinate "Coordinate2D",
+    lower_right_coordinate "Coordinate2D"
+);
+
+CREATE TYPE "RasterBandDescriptor" AS (
+    "name" text,
+    measurement "Measurement"
+);
+
+CREATE TYPE "RasterResultDescriptor" AS (
+    data_type "RasterDataType",
+    -- SpatialReferenceOption
+    spatial_reference "SpatialReference",
+    bands "RasterBandDescriptor" [],
+    "time" "TimeInterval",
+    bbox "SpatialPartition2D",
+    resolution "SpatialResolution"
+);
+
+CREATE TYPE "GdalDatasetGeoTransform" AS (
+    origin_coordinate "Coordinate2D",
+    x_pixel_size double precision,
+    y_pixel_size double precision
+);
+
+CREATE TYPE "FileNotFoundHandling" AS ENUM (
+    'NoData',
+    'Error'
+);
+
+CREATE TYPE "RasterPropertiesKey" AS (
+    domain text,
+    key text
+);
+
+CREATE TYPE "RasterPropertiesEntryType" AS ENUM (
+    'Number',
+    'String'
+);
+
+CREATE TYPE "GdalMetadataMapping" AS (
+    source_key "RasterPropertiesKey",
+    target_key "RasterPropertiesKey",
+    target_type "RasterPropertiesEntryType"
+);
+
+CREATE DOMAIN "StringPair" AS text [2];
+
+CREATE TYPE "GdalRetryOptions" AS (
+    max_retries bigint
+);
+
+CREATE TYPE "GdalDatasetParameters" AS (
+    file_path text,
+    rasterband_channel bigint,
+    geo_transform "GdalDatasetGeoTransform",
+    width bigint,
+    height bigint,
+    file_not_found_handling "FileNotFoundHandling",
+    no_data_value double precision,
+    properties_mapping "GdalMetadataMapping" [],
+    gdal_open_options text [],
+    gdal_config_options "StringPair" [],
+    allow_alphaband_as_mask boolean,
+    retry "GdalRetryOptions"
+);
+
+CREATE TYPE "GdalLoadingInfoTemporalSlice" AS (
+    time "TimeInterval",
+    params "GdalDatasetParameters",
+    cache_ttl int
+);
+
+CREATE TYPE "GdalMetaDataList" AS (
+    result_descriptor "RasterResultDescriptor",
+    params "GdalLoadingInfoTemporalSlice" []
+);
+
+
+
+-- TABLE DEFINITIONS
+
+CREATE TABLE overviews (
```

Comment:
could we not put the overview stuff in the same database schema as the geoengine? Then we wouldn't need two differnent schemas and migrations.

Reply:
How do we provide space in the schema, have no clashes and the overall migration knows what to do there? In theory, there could me multiple providers for ebvs.